examples/hardhat/contracts/Vigil.sol

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+
+contract Vigil {
+    struct SecretMetadata {
+        address creator;
+        string name;
+        /// @notice How long (in seconds) the secret should remain so past the creator's last update.
+        uint256 longevity;
+    }
+
+    event SecretCreated(
+        address indexed creator,
+        string indexed name,
+        uint256 index
+    );
+    event SecretRevealed(
+        address indexed creator,
+        string indexed name,
+        uint256 index
+    );
+
+    SecretMetadata[] public _metas;
+    bytes[] private _secrets;
+    /// @dev The unix timestamp at which the address was last seen.
+    mapping(address => uint256) public _lastSeen;
+
+    function createSecret(
+        string calldata name,
+        uint256 longevity,
+        bytes calldata secret
+    ) external {
+        _updateLastSeen();
+        _metas.push(
+            SecretMetadata({
+                creator: msg.sender,
+                name: name,
+                longevity: longevity
+            })
+        );
+        _secrets.push(secret);
+        emit SecretCreated(msg.sender, name, _metas.length - 1);
+    }
+
+    /// @notice Reveals the secret at the specified index.
```

Comment:
`@notice` is assumed [as default](https://docs.soliditylang.org/en/latest/natspec-format.html#tags) and natspec prefers imperative mood afaik.

```suggestion
    /// Reveal the secret at the specified index.
```